provider/core/src/varule_traits.rs

```diff
@@ -0,0 +1,121 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use zerovec::ule::VarULE;
+
+#[cfg(feature = "alloc")]
+use zerovec::{maps::ZeroMapKV, ZeroMap2d};
+
+/// A trait that associates a [`VarULE`] type with a data struct.
+///
+/// Some data structs can be represented compactly as a single [`VarULE`],
+/// such as `str` or a packed pattern. This trait allows for data providers
+/// to use storage optimizations for such types.
+///
+/// ❗ Not all data structs benefit from this optimization. It is best when the
+/// data struct is multiplied across a large number of data marker attributes.
```

Comment:
this sounds like multiplying across a large number of data marker attributes is an alternative to this optimisation
```suggestion
/// ❗ Not all data structs benefit from this optimization. It works best when the
/// data struct is multiplied across a large number of data marker attributes.
```